privacyidea/lib/resolvers/SQLIdResolver.py

```diff
@@ -367,32 +376,41 @@ def loadConfig(self, config):
                   'Server': self.server,
                   'Database': self.database}
         self.connect_string = self._create_connect_string(params)
+
+        # get an engine from the engine registry, using self.getResolverId() in the key,
+        # which involves the connect string and the pool settings.
+        engine_key = (self.__class__, self.getResolverId())
```

Comment:
Why would not self.getResolverId simply work?

Reply:
This is a remnant from when I thought we could use just the resolver name as the key, and "sqlaudit" as the key for the SQL audit engine. But then, one could name the SQL resolver ``"sqlaudit"`` and have a conflict. So I added the class to the key in order to differentiate between a SQLAudit engine and a SQLIdResolver engine.

But since we now have the unique resolver ID, we could also just use the resolver ID as the key here, and ``"sqlaudit"`` as the key for the SQL audit engine.

Reply:
Your engine_key is a tuple. So it really puzzles me, what the result would be?

Could we simple change the engine_key to self.getResolverId() for the sake of not confusing me and other readers?

Reply:
Well, the engine key is just a dictionary key, and tuples of hashable objects (like class objects, or strings) can be used just fine as dictionary keys. But I agree, as we have the improved ``getResolverId()`` now, this is over-complicating things without any advantage :-) So I'll change the engine keys to simple strings.